### PR TITLE
doc: event propagation for widgets

### PR DESCRIPTION
**Pull request by houqp**

### CODE REVIEW

File: doc/Events.md [hunk 1]
```diff
@@ -1,10 +1,45 @@
-Builtin Events
-==============
+Events
+======
 
-Reader Events
--------------
+## Overview ##
+
+All widgets are subclass of @{ui.widget.eventlistener}, therefore inherit
```
Comment (Frenzie) [inline doc/Events.md]: a subclass

File: doc/Events.md [hunk 2]
```diff
@@ -1,10 +1,45 @@
...
 
 * UpdatePos: emitted by typesetting related modules to notify other modules to
 recalculate the view based on the new typesetting.
 
 * PosUpdate: emitted by readerrolling module to signal a change in pos.
+
+
+## Event propagation ##
+
+Most of the UI components are subclass of
```
Comment (Frenzie) [inline doc/Events.md]: a subclass

File: frontend/ui/widget/container/widgetcontainer.lua [hunk 3]
```diff
@@ -1,7 +1,16 @@
 --[[--
-WidgetContainer is a container for another Widget. Base class for all the widget containers.
+WidgetContainer is a container for one or multiple Widgets. It is the base
+class for all the container widgets.
 
-It handles event propagation and paiting (with different alignments) for its children.
+Children widgets are stored in WidgetContainer as conventional array items:
```
Comment (Frenzie) [inline frontend/ui/widget/container/widgetcontainer.lua]: child widgets

File: frontend/ui/widget/container/widgetcontainer.lua [hunk 4]
```diff
@@ -85,8 +94,8 @@ end
 
 --[[--
 WidgetContainer will pass event to its children by calling their handleEvent
-methods. If no child responded to the event (by returning true), it will call its own
-handleEvent method.
+methods. If no child responded to the event (by returning true), it will try
+consume the event by itself.
```
Comment (Frenzie) [inline frontend/ui/widget/container/widgetcontainer.lua]: to consume

Is consume the right word though? Is it trying to act on the event or is it trying to stop propagation (=consume)?

File: frontend/ui/widget/widget.lua [hunk 5]
```diff
@@ -1,34 +1,40 @@
-local EventListener = require("ui/widget/eventlistener")
-
---[[
-This is a generic Widget interface
+--[[--
+This is a generic Widget interface, which is the base class for all other widgets.
 
-widgets can be queried about their size and can be paint.
+Widgets can be queried about their size and can be paint.
```
Comment (Frenzie) [inline frontend/ui/widget/widget.lua]: can be painted?

File: frontend/ui/widget/widget.lua [hunk 6]
```diff
@@ -1,34 +1,40 @@
...
---[[
-Use this method to define a class that's inherited from current class.
-It only setup the metabale (or prototype chain) and will not initiatie
-a real instance, i.e. call self:init()
---]]
-function Widget:extend(from_o)
-    local o = from_o or {}
+--[[--
+Use this method to define a subclass widget class that's inherited from a
+supper class widget. It only setups the metabale (or prototype chain) and will
```
Comment (Frenzie) [inline frontend/ui/widget/widget.lua]: supper class widget? I'm guessing super class? =P

File: frontend/ui/widget/widget.lua [hunk 7]
```diff
@@ -1,34 +1,40 @@
...
+not initiatie a real instance, i.e. call self:init().
```
Comment (Frenzie) [inline frontend/ui/widget/widget.lua]: initiate

**Comment by houqp** [comment]
Thanks for the catches, all fixed ;)

**Pull request merged**